test/funnels/Depositor.t.sol

```diff
@@ -0,0 +1,595 @@
+            gem1: USDC,
+            fee: uint24(100),
+            tickLower: REF_TICK-100,
+            tickUpper: REF_TICK+100,
+            liquidity: 0,
+            amt0Desired: 2 * cap0,
+            amt1Desired: 2 * cap1,
+            amt0Min: 0,
+            amt1Min: 0
+        });
+
+        vm.expectRevert("Depositor/exceeds-cap");
+        vm.prank(FACILITATOR); depositor.deposit(dp);
+    }
+
+    function testDepositExceedingSlippage() public {
+        Depositor.LiquidityParams memory dp = Depositor.LiquidityParams({
+            gem0: DAI,
+            gem1: USDC,
+            fee: uint24(100),
+            tickLower: REF_TICK-100,
+            tickUpper: REF_TICK+100,
+            liquidity: 0,
+            amt0Desired: 500 * WAD,
+            amt1Desired: 500 * 10**6,
+            amt0Min: 3 * 500 * WAD,
+            amt1Min: 3 * 500 * 10**6
+        });
+
+        vm.expectRevert("Depositor/exceeds-slippage");
+        vm.prank(FACILITATOR); depositor.deposit(dp);
+    }
+
+    function testWithdrawWrongGemOrder() public {
+        Depositor.LiquidityParams memory dp = Depositor.LiquidityParams({
+            gem0: USDC,
+            gem1: DAI,
+            fee: uint24(100),
+            tickLower: REF_TICK-100,
+            tickUpper: REF_TICK+100,
+            liquidity: 0,
+            amt0Desired: 0,
+            amt1Desired: 0,
+            amt0Min: 0,
+            amt1Min: 0
+        });
+
+        vm.expectRevert("Depositor/wrong-gem-order");
+        vm.prank(FACILITATOR); depositor.withdraw(dp, false);
+    }
+
+    function testWithdrawTooSoon() public {
+        Depositor.LiquidityParams memory dp = Depositor.LiquidityParams({
+            gem0: DAI,
+            gem1: USDC,
+            fee: uint24(100),
+            tickLower: REF_TICK-100,
+            tickUpper: REF_TICK+100,
+            liquidity: 0,
+            amt0Desired: 500 * WAD,
+            amt1Desired: 500 * 10**6,
+            amt0Min: 490 * WAD,
+            amt1Min: 490 * 10**6
+        });
+
+        vm.prank(FACILITATOR); (uint128 liq,,) = depositor.deposit(dp);
+        dp.liquidity = liq;
+
+        vm.expectRevert("Depositor/too-soon");
+        vm.prank(FACILITATOR); depositor.withdraw(dp, false);
+    }
+
+    function testWithdrawNoPosition() public {
+        Depositor.LiquidityParams memory dp = Depositor.LiquidityParams({
+            gem0: DAI,
+            gem1: USDC,
+            fee: uint24(100),
+            tickLower: REF_TICK-100,
+            tickUpper: REF_TICK+100,
+            liquidity: 1,
+            amt0Desired: 0,
+            amt1Desired: 0,
+            amt0Min: 0,
+            amt1Min: 0
+        });
+
+        // "Liquidity Sub" error - https://github.com/Uniswap/v3-core/blob/d8b1c635c275d2a9450bd6a78f3fa2484fef73eb/contracts/libraries/LiquidityMath.sol#L12
+        vm.expectRevert(bytes("LS"));
+        vm.prank(FACILITATOR); depositor.withdraw(dp, false);
+    }
+
+    function testWithdrawExceedingCap() public {
```

Comment:
Same for each cap

Reply:
Done in [f23d585](https://github.com/makerdao/dss-allocator/pull/2/commits/f23d585b25acdf5f855375d4d67f5ec37a07aee0)